### PR TITLE
Remove sample if not going through seqlab

### LINKED ISSUE #70: Route samples after "Awaiting User Response" to Remove Step
At the end of "Awaiting User Response" artifacts that do not have UDF "Proceed To SeqLab" == true
Should be routed and run through the Remove from workflow step.
this can be done in `assign_workflow_preseqlab.py` or another independent script

### PR DESCRIPTION
**Pull request by tcezard**
Change `assign_workflow_preseqlab.py` to push samples through the remove workflow when the UDF `Proceed To SeqLab` is not set.
fixes #70

### CODE REVIEW

File: scripts/assign_workflow_preseqlab.py [hunk 1]
```diff
@@ -1,28 +1,55 @@
...
+
+        if artifacts_to_route_to_remove:
+            stage = get_workflow_stage(self.lims, "Remove From Processing EG 1.0 WF", "Remove From Processing EG 1.0 ST")
+            self.lims.route_artifacts(list(artifacts_to_route_to_remove), stage_uri=stage.uri)
+
+            # Create new step with the routed artifacts
+            s = Step.create(self.lims, protocol_step=stage.step, inputs=artifacts_to_route_to_remove,
+                            container_type_name='Tube')
+
+            s.details.udf['Reason for removal from processing:'] = 'Failed QC. See step id %s' % self.process.id
```
Comment (EGCGDanForth) [inline scripts/assign_workflow_preseqlab.py]: Could this return the uri rather than just the id?

Comment (tcezard) [reply scripts/assign_workflow_preseqlab.py]: Done

File: scripts/assign_workflow_preseqlab.py [hunk 2]
```diff
@@ -1,28 +1,79 @@
...
+    def _finish_step(step, try_count=1):
+        """
+        This function will try to advance a step three time waiting for a ongoing program to finish.
+        It waits for 5 seconds in between each try
+        """
+        try:
+            step.get(force=True)
+            step.advance()
+        except HTTPError as e:
+            if try_count < 3 and str(e) == '400: Cannot advance a step that has an external program queued, ' \
```
Comment (mwhamgenomics) [inline scripts/assign_workflow_preseqlab.py]: I think this will only try twice. Might be good to add a test for this as well

Comment (tcezard) [reply scripts/assign_workflow_preseqlab.py]: No it will try  three time as it is.
I've added test now

**Pull request merged**